src/content/docs/guides/development-workflow.mdx

```diff
@@ -0,0 +1,145 @@
+---
+
+import RemoteCode from '~/components/RemoteCode.astro';
+import CICheckSnippet from '~/components/guides/development-workflow/CICheckSnippet.astro';
+
+This guide will walk through an opinionated development workflow with Shorebird.
+We'll cover how to automate releasing and patching with Shorebird to streamline
+the entire development process so that you and your team can focus on shipping
+high quality features to customers.
+
+## Prerequisites
+
+This guide assumes you are already familiar with Shorebird and have integrated
+it into your application. Refer to the [getting started](/) instructions for
+more information.
+
+## Overview
+
+Shorebird provides a set of tools that you can use to push over the air updates
+to your customers but it's up to you to decide how to integrate it into your
+development workflow.
+
+While there are many ways to integrate Shorebird, the following workflow is a
+common approach that we recommend:
+
+![development-workflow](https://github.com/user-attachments/assets/59eaefc4-1213-4549-b0d6-38738b065147)
+
+The illustration above, uses [trunk based
+development](https://trunkbaseddevelopment.com) and tags to trigger automated
+release and patch workflows.
+
+The workflow consists of three main phases:
+
+1. **Development**: Developers work on features and bug fixes.
+1. **Release**: When a feature is ready, a release is created and distributed to
+   customers (typically via stores).
+1. **Patch**: If a critical bug is found, a patch is created and distributed to
+   customers immediately via Shorebird.
+
+Let's walk through each phase in more detail.
+
+## Development Phase
+
+Most of the time is spent in the development phase. During this phase,
+developers are building features and fixing bugs.
+
+The development workflow can be broken down into the following steps:
+
+1. Create branch from trunk (main)
+1. Work on task
+1. Open a pull request to merge changes into trunk
+1. Squash and merge after ci checks & code review
+
+Note that with the described workflow, the trunk (e.g. main) is always in a
+releasable state. This puts pressure on continuous integration (CI) checks to ensure
+that the code on the trunk is always in a healthy, deployable state. Checks that would
+typically run as part of the continuous integration process include:
+
+- Linting
+- Formatting
+- Unit tests
+- Integration tests
+- Test coverage reports
+
+You can refer to the following GitHub Actions workflow for an example.
+
+<CICheckSnippet />
+
+## Release Phase
+
+When the team is ready to distribute a new version of the app, a release is
+created. In some cases, releases are created on a regular cadence (e.g. weekly,
+bi-weekly, monthly) while in other cases, releases are created on an ad-hoc
+basis.
+
+The release workflow can be broken down into the following steps:
+
+1. Create a release branch (e.g release/v1.0.0) from trunk (main)
```

Comment:
Should we lean on Github releases for this?

Reply:
Yup I was just spelling out the explicit steps but yeah I can reword to leverage releases